website_compassion/models/payment_transaction.py

```diff
@@ -0,0 +1,19 @@
+##############################################################################
+#
+#    Copyright (C) 2021 Compassion CH (http://www.compassion.ch)
+#    @author: Robin Berguerand <robin.berguerand@gmail.com>
+#
+#    The licence is in the file __manifest__.py
+#
+##############################################################################
+from odoo import models, fields
+
+
+class PaymentTransaction(models.Model):
+    _inherit = "payment.transaction"
+
+    def _set_transaction_cancel(self):
+        for invoice in self.invoice_ids.filtered(lambda a: a.cancel_invoice):
```

Comment:
If the filter function is directly on a field, you can simplify and avoid using a lambda.
```suggestion
        for invoice in self.invoice_ids.filtered('auto_cancel_no_transaction'):
```